src/avram/callbacks.cr

```diff
@@ -80,6 +104,55 @@ module Avram::Callbacks
     end
   end
 
+  macro after_run(method_name)
+    def after_run(object)
+      {% if @type.methods.map(&.name).includes?(:after_run.id) %}
+        previous_def
+      {% else %}
+        super
+      {% end %}
+
+      {{ method_name.id }}(object)
+    end
+  end
+
+  # Run the given block after the operation runs
+  #
+  # The return value from `run` will be passed to this block.
+  #
+  # ```
+  # class GenerateReport < Avram::Operation
+  #   after_run do |value|
+  #     value == "some report"
+  #   end
+  #
+  #   def run
+  #     "some report"
+  #   end
+  # end
+  # ```
+  macro after_run(&block)
+    {%
+      if block.args.size != 1
+        raise <<-ERR
+        The 'after_run' callback requires only 1 block arg to be passed.
+        Example:
+          after_run { |value| some_method(value) }
+        ERR
+      end
+    %}
+    def after_run(object)
+      {% if @type.methods.map(&.name).includes?(:after_run.id) %}
+        previous_def
+      {% else %}
+        super
+      {% end %}
+
+      {{ block.args.first }} = object
+      {{ block.body }}
```

Comment:
in the before run I think the code actually yields to the given block but this inlines it. Should it be consistent between the two?

Reply:
No, because this is a macro yield and the other is an method yield. These actually have to work different.